core/example/benchmark/CMakeLists.txt

```diff
@@ -8,3 +8,9 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "([xX]86|[aA][mM][dD]64)")
   add_executable(PeakFlops Cabana_peakflops.cpp)
   target_link_libraries(PeakFlops cabanacore)
 endif()
+
+if(Cabana_ENABLE_MPI AND Cabana_ENABLE_Cuda AND (Cabana_ENABLE_Serial OR Cabana_ENABLE_OpenMP OR Cabana_ENABLE_Pthread) )
```

Comment:
I don't get the `AND (Cabana_ENABLE_Serial OR Cabana_ENABLE_OpenMP OR Cabana_ENABLE_Pthread)`

Reply:
Essentially this should only build if Cuda is on and an execution space is enabled that supports `Kokkos::HostSpace`. Are there configurations of Kokkos possible where `HostSpace` is not supported?

Reply:
You only use the memory space and it will be there even when CUDA only is enabled

Reply:
Per phone call we need to do a follow-on PR which templates our classes on device types instead of memory spaces. Then we would refactor this to explicitly need `Cuda` and `OpenMP`. Once we do that then this test will specify device types that have both memory and execution space definitions.